vault/logical_system_custom_messages.go

```diff
@@ -426,24 +426,16 @@ func (b *SystemBackend) handleCreateCustomMessages(ctx context.Context, req *log
 	if linkMap != nil {
 		link = &uicustommessages.MessageLink{}
```

Comment:
🤔 The original structure seems a bit more extensible as well. We can always add additional keys but they might look odd adjacent to the inferred title from the key in the `link` map. For example:

```
{
  "link": {
    "Click Here": "https://www.example.com",
    "description": "Lorem ipsum dolor sit amet"
  }
}
```

Reply:
Hmm, that's a good point -- though, something I might be missing: what was the impetus to change the format?

Reply:
That request came from the UI team.

Reply:
To @VioletHynes 's earlier comment:
I brought this up while discussing this aspect of the feature. For now, the intent from the UI is to limit this to a single link. I think that the current API contract is well positioned for the possibility of a change in this stance to support multiple links (_i.e._ link is a JSON object). We would need to make a change to the **uicustommessages.Message** struct to hold a slice of **uicustommessages.MessageLink** instead of a single pointer to one.

Reply:
`link` is a JSON object, but its contents are arbitrary, so it would be hard to modify or add to the if we go with this approach. I think that's probably okay, though, and far from a show-stopper. If we want to support multiple in the future, we could add a `links` field to the JSON or similar, so we can extend it if needs be.

I do think, since we only support one link, we should add validation to `linkMap` to ensure that there is only one link, since that's all we expect the UI to send us, and anybody directly using the API shouldn't (unless we enhance the feature) be trying to specify two links.

Reply:
Ah, I see. Do you mean verifying that the `map[string]any` in the **framework.FieldData** (for key `link`) only contains one element? Since if there are more than one, there's no way of knowing which one will be considered and stored with the message.